gui/src/main/java/cern/jarrace/inspector/gui/rest/ContainerService.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * © Copyright 2016 CERN. This software is distributed under the terms of the Apache License Version 2.0, copied
+ * verbatim in the file “COPYING”. In applying this licence, CERN does not waive the privileges and immunities granted
+ * to it by virtue of its status as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package cern.jarrace.inspector.gui.rest;
+
+import cern.jarrace.commons.domain.AgentContainer;
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+import java.util.List;
+
+/**
+ * A service that can access a remote endpoint with methods related to {@link AgentContainer}s.
+ */
+public interface ContainerService {
```

Comment:
Hmm I'm not sure to follow you.
The fact that we use a common interface and/or common code now does not mean that later we wont be able to have a client written in a different language with different technologies (granted the server remains REST).
And there are advantages now to re use the same code.